### PR TITLE
Tdd

### PR DESCRIPTION
**Pull request by emiliazm**
Hello!
### Project requirements
- Create a class called `Solver`.
- Create a method called `factorial` that takes one argument, an integer `N`, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to `N` and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
- Create a method called `reverse` that takes one argument, a string `word`, and returns `word` reversed (e.g. if `word` is `"hello"` it returns `"olleh"`).
- Create a method called `fizzbuzz` that takes one argument, an integer `N`, and returns a string. The returned string is constructed following these rules:
  - When `N` is divisible by 3, return `"fizz"`.
  - When `N` is divisible by 5, return `"buzz"`.
  - When `N` is divisible by 3 and 5, return `"fizzbuzz"`.
  - Any other case, return `N` as a string (e.g. say `N` is `7` then return `"7"`).
- Remember to start with tests for all of the methods described above. Your commit history should prove that you have used TDD.

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+require_relative './../solver'
+
+describe 'Solver' do
+  describe 'When testing factorial' do
+    it 'Factorial test' do
+      number = 5
+      solver = Solver.new
+      factorial = solver.factorial(number)
+      expect(factorial).to eq(120)
+    end
```
Comment (mikethreels) [inline spec/solver_spec.rb]: - [ ] for the factorial method you should also write tests for negative numbers and 0.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by emiliazm** [comment]
Hello @mikethreels !
Thank you for your comments!

### These are the changes that we implemented:
- For the factorial method you should also write tests for negative numbers and 0.

**Review by ahmetbozaci — APPROVED** [review]
# 🥳 🎉🥇 💯  💥 🎆 🎈 STATUS: APPROVED 🥳 🎉  🥇  💯  💥   🎆 🎈 🎉 



@saifullah767 @emiliazm  ✋🏼 
#### Your project is great. 🎉 I approve of it. There is nothing else to say other than... it's time to merge it. You can submit this project on your dashboard.
___ 

### 💯 Highlights 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
- All tests are passed.✔️

___ 

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
aproved


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_
